aea/skills/base.py

```diff
@@ -389,18 +389,70 @@ def parse_module(
 class Task(SkillComponent):
     """This class implements an abstract task."""
 
-    def __init__(self, *args, **kwargs):
+    def __init__(
+        self,
+        name: Optional[str] = None,
+        skill_context: Optional[SkillContext] = None,
+        **kwargs
+    ):
         """
         Initialize a task.
 
-        :param skill_context: the skill context
+        :param name: the name of the task (mandatory)
+        :param skill_context: the skill context (mandatory)
         :param kwargs: keyword arguments.
         """
-        super().__init__(**kwargs)
-        self.completed = False
+        super().__init__(name=name, skill_context=skill_context, **kwargs)
+
+        self._executed = False
```

Comment:
the "functor" pattern is used.

`executed` says whether the function has already been executed.

Reply:
you should mention the `functor` patterns in the docs